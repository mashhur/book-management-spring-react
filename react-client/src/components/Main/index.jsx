import React, { useState } from "react";
import { useParams } from "react-router";
import Books from "../Books";
import CategoryCrumb from "../CategoryCrumb";

function Main() {
	const { category } = useParams();
	return (
		<main
			style={{
				height: "89vh",
				overflowY: "scroll",
				background: "#f9f9f9a6",
			}}
			className="col-md-9   ">
			<CategoryCrumb category={category} />
			<Books category={category ? category : "all"} />
		</main>
	);
}

export default Main;
