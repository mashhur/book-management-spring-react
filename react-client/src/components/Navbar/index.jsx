import react from "react";
import { Link } from "react-router-dom";
import { TiShoppingCart } from "react-icons/ti";
import { useSelector } from "react-redux";
function Navbar() {
	const user = useSelector((state) => state.user);
	return (
		<nav className="navbar navbar-expand-lg border-bottom  navbar-light bg-light">
			<div className="container-fluid">
				<Link className="navbar-brand" to="/">
					<p
						style={{ opactity: ".5" }}
						className="fs-2 mb-0 ">
						<span style={{ letterSpacing: "-1px" }}>
							Diamond
						</span>
						<small
							style={{
								fontSize: "18px",
								letterSpacing: "2px",
							}}>
							<b>BOOKS</b>
						</small>
					</p>
				</Link>
				<button
					className="navbar-toggler"
					type="button"
					data-toggle="collapse"
					data-target="#navbarNav"
					aria-controls="navbarNav"
					aria-expanded="false"
					aria-label="Toggle navigation">
					<span className="navbar-toggler-icon"></span>
				</button>
				<div className="w-100 d-flex justify-content-center">
					<input
						className="form-search  px-4 w-50 mx-5"
						type="search"
						placeholder="Search for a book, author, or category"
						aria-label="Search"
					/>
				</div>
				<div
					className="collapse navbar-collapse"
					id="navbarNav">
					<ul className="navbar-nav ms-auto">
						<li className="nav-item ">
							<Link className="nav-link" to="/cart">
								<div className="position-relative">
									<TiShoppingCart
										style={{
											fontSize: "40px",
											opacity: 0.8,
										}}
									/>
									{user.cart.length > 0 && (
										<span
											data-test="cart-item-count"
											className="position-absolute top-0 start-100 translate-middle badge rounded-pill bg-success">
											{user.cart.length}
										</span>
									)}
								</div>
							</Link>
						</li>
						<li className="nav-item active">
							<Link className="nav-link" to="/profile">
								<img
									style={{ height: "45px", width: "45px" }}
									src="https://www.w3schools.com/howto/img_avatar2.png"
									className="rounded-circle"
									alt="avatar"
								/>
							</Link>
						</li>
					</ul>
				</div>
			</div>
		</nav>
	);
}
export default Navbar;
