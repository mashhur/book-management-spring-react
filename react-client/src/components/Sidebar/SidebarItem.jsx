import { BiChevronRight } from "react-icons/bi";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";

function SidebarItem({ name, url }) {
	return (
		<div
			data-test="component-side-bar-item"
			style={{
				borderRadius: "2.3rem",
				fontSize: "1.0rem",
				marginBottom: ".5rem",
			}}
			className="col">
			<Link
				to={url ? url : `/cat/${name}`}
				className="list-group-item d-flex 
				justify-content-between 
				align-items-center 
				list-group-item-action 
				py-3 px-4 bg-category fs-6 ">
				{name}
				<span>
					<BiChevronRight
						style={{ fontSize: "23px", opacity: 0.6 }}
					/>
				</span>
			</Link>
		</div>
	);
}

SidebarItem.propType = {
	name: PropTypes.string.isRequired,
	url: PropTypes.string,
};

export default SidebarItem;
