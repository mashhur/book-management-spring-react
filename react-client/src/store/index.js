import { configureStore } from "@reduxjs/toolkit";
import locationReducer from "./slices/locationSlice";
import userReducer from "./slices/userSlice";

export const store = configureStore({
	reducer: {
		location: locationReducer,
		user: userReducer,
	},
});
