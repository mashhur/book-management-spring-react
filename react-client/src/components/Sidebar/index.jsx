import React, { useEffect, useState } from "react";
import { Link } from "react-router-dom";
import CategoryService from "../../service/CategoryService";
import { BiChevronRight } from "react-icons/bi";
import SidebarItem from "./SidebarItem";

function Sidebar() {
	const [categories, setcategories] = useState([]);

	useEffect(() => {
		CategoryService.getAllCategories().then((res) => {
			setcategories(res.data);
		});
	}, []);

	return (
		<aside
			data-test="component-side-bar"
			style={{
				overflowY: "scroll",
				paddingBottom: "2rem",
				height: "89vh",
				background: "#f9f9f9a6",
			}}
			className="col-md-3 border px-0">
			<h4 className="h5 text-muted py-4 px-3 mb-3 border-bottom">
				Categories
			</h4>
			<p className="w-75 px-4 fs-6 text-muted">
				A curated list of every book ever written
			</p>

			<div className=" px-2  ">
				<div>
					<SidebarItem name="All Books" url="/" />

					{categories.map((category) => (
						<SidebarItem
							name={category.name}
							key={category.id}
						/>
					))}
				</div>
			</div>
		</aside>
	);
}

export default Sidebar;
