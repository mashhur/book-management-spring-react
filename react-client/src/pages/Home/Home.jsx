import React, { useEffect } from "react";
import Main from "../../components/Main";
import Sidebar from "../../components/Sidebar";
import Navbar from "../../components/Navbar";
function Home() {
	return (
		<>
			<Navbar />
			<div
				className="container-fluid"
				style={{ overflow: "hidden" }}>
				<section className="row ">
					<Sidebar />
					<Main />
				</section>
			</div>
		</>
	);
}

export default Home;
