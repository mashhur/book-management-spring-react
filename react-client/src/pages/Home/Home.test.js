import React from "react";
import { mount } from "enzyme";
import Home from "./Home";
import Navbar from "../../components/Navbar";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { createTestStore } from "../../store";
import Sidebar from "../../components/Sidebar";
import Main from "../../components/Main";
const store = createTestStore();
const setup = () => {
	return mount(
		<BrowserRouter>
			<Provider store={store}>
				<Home />
			</Provider>
		</BrowserRouter>
	);
};

describe("Testing books component", () => {
	it("renders without any error", async () => {
		const wrapper = setup();
		expect(wrapper.find(Navbar).exists()).toBe(true);
	});

	it("renders `Navbar` by default", async () => {
		const wrapper = setup();
		expect(wrapper.find(Navbar).exists()).toBe(true);
	});

	it("renders `Sidebar` by default", async () => {
		const wrapper = setup();
		expect(wrapper.find(Sidebar).exists()).toBe(true);
	});

	it("renders `Main` by default", async () => {
		const wrapper = setup();
		expect(wrapper.find(Main).exists()).toBe(true);
	});
});
