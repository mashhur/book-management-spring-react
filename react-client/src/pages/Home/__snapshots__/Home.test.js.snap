// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing books component should match the default snapshot 1`] = `ReactWrapper {}`;
